hub/util/shape_interval.py

```diff
@@ -9,11 +9,13 @@ def _contains_negatives(shape: Sequence[int]):
 
 
 class ShapeInterval:
-    def __init__(self, lower: Sequence[int], upper: Sequence[int] = None):
+    def __init__(self, lower: Sequence[int], upper: Optional[Sequence[int]] = None):
```

Comment:
is it possible to get None `upper` interval?

Reply:
This change was made because `upper` is being set to `None` here. Don't have much context on it but it looks like upper can be `None`